src/Xenial.Tasty.Cli/Internal/TastyCommander.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.Threading;
+
+using static SimpleExec.Command;
+
+namespace Xenial.Delicious.Cli.Internal
+{
+    internal class TastyCommander
+    {
+        public async Task<int> BuildProject(string csProjFileName, IProgress<(string line, bool isRunning, int exitCode)> progress, CancellationToken cancellationToken = default)
+        {
+            return await Task.Run(async () =>
+            {
+                var buildTask = BuildAsync(csProjFileName, cancellationToken);
+
+                await foreach (var (line, isRunning, exitCode) in buildTask)
+                {
+                    progress.Report((line, isRunning, exitCode));
+                    if (!isRunning)
+                    {
+                        return exitCode;
+                    }
+                }
+                return 0;
+            });
+        }
+
+        public static async IAsyncEnumerable<(string, bool, int)> BuildAsync(string csProjFileName, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```

Comment:
bool param is always true
maybe give the tuple parts names

Reply:
It should go to false if the process stops (which it doesn't at the moment, which is wrong)
I better rename it to `hasStopped` that will make the intent clearer.